blockchains/xrp/xrp_worker.js

```diff
@@ -51,23 +53,33 @@ class XRPWorker extends BaseWorker {
     this.lastConfirmedBlock = parseInt(lastValidatedLedgerData.ledger.ledger_index) - constants.CONFIRMATIONS;
   }
 
-  async fetchLedgerTransactions(connection, ledger_index) {
-    const ledger = await this.connectionSend(connection, {
+  async fetchLedger(connection, ledger_index, should_expand) {
+    const result = await this.connectionSend(connection, {
       command: 'ledger',
       ledger_index: parseInt(ledger_index),
       transactions: true,
-      expand: false
-    }).then(value => value.result.ledger);
+      expand: should_expand
+    });
 
-    assert(ledger.closed === true);
+    const ledger = result.result.ledger;
 
-    if (typeof ledger.transactions === 'undefined' || ledger.transactions.length === 0) {
-      return { ledger: ledger, transactions: [] };
+    assert(ledger.closed === true);
```

Comment:
Isn't using assertions in non-testing code discouraged? It's not really elegant in error-handling terms